shadows/framework/src/main/java/org/robolectric/shadows/ShadowTelephonyManager.java

```diff
@@ -225,25 +224,25 @@ public int getEventFlags() {
   }
 
   @Implementation(minSdk = S)
-  public void registerTelephonyCallback(Executor executor, TelephonyCallback callback) {
+  public void registerTelephonyCallback(/*Executor*/ Object executor, /*TelephonyCallback*/ Object callback) {
```

Comment:
Could you do a type checking in `registerTelephonyCallback` with `Preconditions.checkArgument` + `instanceof`? `ShadowAppOpsManager#noteProxyOpNoThrow` has an example to use it.

Reply:
Sure, please review again